README.md

```diff
@@ -198,6 +198,36 @@ buildctl build ... --exporter=oci --exporter-opt output=path/to/output.tar
 buildctl build ... --exporter=oci > output.tar
 ```
 
+### Exporting/Importing build cache (not image itself)
+
+#### To/From registry
+
+```
+buildctl build ... --cache-exporter registry --cache-exporter-opt ref=localhost:5000/myrepo:buildcache
+buildctl build ... --cache-importer registry --cache-importer-opt ref=localhost:5000/myrepo:buildcache
+```
+
+#### To/From local filesystem
+
+```
+buildctl build ... --cache-exporter local --cache-exporter-opt output=path/to/output-dir
+buildctl build ... --cache-importer local --cache-importer-opt input=path/to/input-dir --cache-importer-opt digest=sha256:deadbeef
+```
+
+The directory layout conforms to OCI Image Spec v1.0.
+However, `index.json` is not generated, because creating `index.json` would require locking the file, which can be problematic on shared filesystems.
+
+#### `--cache-exporter-opt` options
```

Comment:
The API supports multiple exporters. How would we support this in CLI? Multiple importers are already supported(and somewhat useful). This PR seems to break this.

Reply:
My idea was like 

```
buildctl build \
 --cache-exporter registry --cache-exporter-opt ref=some/registry/ref \
 --cache-exporter local      --cache-exporter-opt output=some/local/dir \
 --cache-exporter registry --cache-exporter-opt ref=yet/another/registry/ref
```
.

But I noticed `urfave/cli` can't support such style, because  `urfave/cli` uses `flag.Parse()` , [which calls `Value.Set()` without passing the original positional info](https://github.com/golang/go/blob/f500f13d72de3ca3201ec2678a7f88524353bede/src/flag/flag.go#L888)  😢

Reply:
If we introduce a comma-separated `refs` attribute (`--cache-importer registry --cache-importer-opt refs=some/registry/ref,yet/another/registry/ref`),  we can continue to support the multi-registry usecase in the new CLI, although it still can't support specifying `registry` and `local` simultaneously. Maybe we should use YAML for such complicated scenario.

Reply:
(Aside from this PR we should also support using multiple image exporters simultaneously)

Reply:
Sorry for going back and forth but can we reconsider using CSV for everything?
```console
buildctl build \
  --frontend type=dockerfile.v0,target=foo,build-arg:foo=bar \
  --export type=image,name=docker.io/username/image,push=true \
  --import-cache type=registry,ref=some/registry/ref \
  --import-cache type=registry,ref=yet/another/registry/ref \
  --export-cache type=registry,ref=some/registry/ref \
  --export-cache type=local,store=some/local/dir
``` 

cc @thaJeztah

Reply:
Just dropping in, but 👍 for using CSV. Yes; it's verbose, but it's non-ambiguous: options that belong together are also grouped together in a single flag.

We could probably still handle shorthands if no options are used (not sure if there's exporters that require no options?), i.e. translate `--cache-exporter registry` to `--cache-exporter type=registry` behind the scenes

Reply:
I guess I'm ok with csv. I don't like `--export type=local` very much but don't have a better solution. Maybe we can do the shorthands like @thaJeztah suggested to avoid the extra typing.

Reply:
updated CLI